propolis/src/block.rs

```diff
@@ -59,140 +62,34 @@ pub trait BlockDev<R: BlockReq>: Send + Sync + 'static {
     fn inquire(&self) -> BlockInquiry;
 }
 
-/// Abstraction over an actual backing store
-pub trait BackingStore: Send + Sync + 'static {
-    /// Read from backing store and write into guest mapping
-    fn issue_read(
-        &self,
-        offset: usize,
-        sz: usize,
-        mapping: SubMapping,
-    ) -> Result<usize>;
-
-    /// Read from guest mapping and write into backing store
-    fn issue_write(
-        &self,
-        offset: usize,
-        sz: usize,
-        mapping: SubMapping,
-    ) -> Result<usize>;
-
-    /// Issue flush command to backing store
-    fn issue_flush(&self) -> Result<()>;
-
-    /// Is backing store read only?
-    fn is_ro(&self) -> bool;
-
-    /// return total length in bytes
-    fn len(&self) -> usize;
-
-    /// some backing stores may have block size requirements. optionally return a
-    /// block size in bytes.
-    fn block_size(&self) -> Option<usize>;
-}
-
-/// A block device implementation backed by an underlying file.
-pub struct FileBackingStore {
+/// Standard [`BlockDev`] implementation.
+pub struct PlainBdev<R: BlockReq> {
```

Comment:
Should we rename this now to indicate it is file backed? `FileBdev` or something. Then as we add others like `CruicibleBdev` it'll be consistent.

Reply:
done in e53f8a3